agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/dsl/builder/AIAgentSubgraphBuilder.kt

```diff
@@ -56,11 +60,29 @@ public abstract class AIAgentSubgraphBuilderBase<Input, Output> {
      * @param nextNode Node to connect to
      * @return The next node
      */
-    public infix fun <IncomingOutput, OutgoingInput, OutgoingOutput> AIAgentNodeBase<IncomingOutput, OutgoingInput>.then(nextNode: AIAgentNodeBase<OutgoingInput, OutgoingOutput>): AIAgentNodeBase<OutgoingInput, OutgoingOutput> {
+    public infix fun <IncomingOutput, OutgoingInput, OutgoingOutput> AIAgentNodeBase<IncomingOutput, OutgoingInput>.then(
+        nextNode: AIAgentNodeBase<OutgoingInput, OutgoingOutput>
+    ): AIAgentNodeBase<OutgoingInput, OutgoingOutput> {
         edge(this forwardTo nextNode)
         return nextNode
     }
 
+    /**
+     * Executes multiple nodes in parallel and merges their contexts.
+     *
+     * @param nodes List of nodes to execute in parallel
+     * @param reduce Function to merge the contexts after parallel execution
+     * @param name Optional name for the parallel node
+     * @return A node that executes the given nodes in parallel
+     */
+    public fun <Input, Output> parallel(
+        vararg nodes: AIAgentNodeBase<Input, Output>,
+        reduce: suspend AIAgentContextBase.(List<Triple<String, AIAgentContextBase, Output>>) -> Pair<AIAgentContextBase, Output>,
```

Comment:
Should I use Triple here or create separate data class like `ParallelNodeResult`?